File: backend/src/models/Cliente.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/database");

const Cliente = sequelize.define(
	"Cliente",
	{
		id_cliente: {
			type: DataTypes.INTEGER,
			primaryKey: true,
			autoIncrement: true,
		},
		id_empresa: {
			// id_tenant - FK hacia empresas
			type: DataTypes.INTEGER,
			allowNull: false,
			references: {
				model: "empresas",
				key: "id_empresa",
			},
			comment: "ID Tenant - Referencia a la microempresa/empresa",
		},
		nombre: {
			type: DataTypes.STRING(200),
			allowNull: false,
		},
		nit: {
			type: DataTypes.STRING(50),
		},
		telefono: {
			type: DataTypes.STRING(20),
		},
		email: {
			type: DataTypes.STRING(100),
		},
		direccion: {
			type: DataTypes.STRING(300),
		},
		activo: {
			type: DataTypes.BOOLEAN,
			defaultValue: true,
		},
	},
	{
		tableName: "clientes",
		timestamps: true,
		paranoid: true, // ✅ ACTIVADO: Soft delete
		createdAt: "fecha_creacion",
		updatedAt: "fecha_actualizacion",
		deletedAt: "fecha_eliminacion", // ✅ ACTIVADO
	},
);

// Definir asociaciones
Cliente.associate = (models) => {
	// Un cliente pertenece a una empresa
	Cliente.belongsTo(models.Empresa, {
		foreignKey: "id_empresa",
		as: "empresa",
	});

	// Un cliente tiene muchas ventas
	Cliente.hasMany(models.Venta, {
		foreignKey: "id_cliente",
		as: "ventas",
	});
};

module.exports = Cliente;

File: mobile/lib/models/cliente.dart
class Cliente {
  final int id;
  final String nombre;
  final String? email;
  final String? telefono;
  final String? direccion;
  final String? documento;
  final String? tipoDocumento;
  final bool activo;
  final int idEmpresa;

  Cliente({
    required this.id,
    required this.nombre,
    this.email,
    this.telefono,
    this.direccion,
    this.documento,
    this.tipoDocumento,
    required this.activo,
    required this.idEmpresa,
  });

  factory Cliente.fromJson(Map<String, dynamic> json) {
    return Cliente(
      id: json['id_cliente'] ?? json['id'],
      nombre: json['nombre'] ?? '',
      email: json['email'],
      telefono: json['telefono'],
      direccion: json['direccion'],
      documento: json['nit'] ?? json['documento'],
      tipoDocumento: json['tipo_documento'],
      activo: json['activo'] ?? true,
      idEmpresa: json['id_empresa'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nombre': nombre,
      'email': email,
      'telefono': telefono,
      'direccion': direccion,
      'documento': documento,
      'tipo_documento': tipoDocumento,
      'activo': activo,
    };
  }
}
